components/suggestion/suggestion_box.jsx

```diff
@@ -463,11 +471,35 @@ export default class SuggestionBox extends React.PureComponent {
 
         this.inputRef.current.focus();
 
-        for (const provider of this.props.providers) {
-            if (provider.handleCompleteWord) {
-                provider.handleCompleteWord(term, matchedPretext, this.handlePretextChanged);
+        let simulateEnter = false;
+        if (e && Utils.isKeyPressed(e, Constants.KeyCodes.TAB)) {
+            e.preventDefault();
+            simulateEnter = true;
+        }
+        if (!e) {
+            simulateEnter = true;
+        }
+
+        let ke = e;
+        if (simulateEnter) {
+            ke = new KeyboardEvent('keydown', {
+                bubbles: true, cancelable: true, keyCode: 13,
+            });
+        }
+
+        if (finish && this.props.onKeyPress) {
```

Comment:
I haven't worked it out, but it feels like this can be simplified. If you're only sending the keyPress if `finish === true`, then all the above only needs to be run on `if (finish && this.props.onKeyPress)`. That should help simplify it.